code/src/components/TaskList.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { removeTask, setAllItems, toggleChecked } from 'reducers/tasks';
+import styled from 'styled-components/macro';
+import removeTaskIcon from 'assets/x-alphabet-icon.svg'
+import { filters } from './TaskFilters';
+
+const TaskList = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    const listFromStorage = JSON.parse(localStorage.getItem('toDoList'))
+    if (listFromStorage) {
+      dispatch(setAllItems(listFromStorage))
+    }
+  }, [dispatch])
+
+  const taskList = useSelector((store) => store.tasks.items);
+  const filter = useSelector((store) => store.tasks.filteredBy);
+
+  // ⬇ This function is filtering the task list based on ALL/COMPLETED/NOTCOMPLETED
+  const filteredTasksList = () => {
+    if (filter === filters.COMPLETED) {
+      return taskList.filter((task) => task.isChecked);
+    } else if (filter === filters.NOT_COMPLETED) {
+      return taskList.filter((task) => !task.isChecked);
+    } else {
+      return taskList;
+    }
+  };
+
+  useEffect(() => {
+    localStorage.setItem('toDoList', JSON.stringify(taskList))
```

Comment:
Well done solving the useEffect so it synchronizes all tasks. Thank you also for showing and sharing this to the rest of the group!